File: SYP.Web/Modules/ServerTypes/Warehouse/StockEntriesColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { StockEntriesRow } from "./StockEntriesRow";
import { StockEntryStatus } from "./StockEntryStatus";

export interface StockEntriesColumns {
    EntryNo: Column<StockEntriesRow>;
    WarehouseName: Column<StockEntriesRow>;
    EntryDate: Column<StockEntriesRow>;
    Status: Column<StockEntriesRow>;
    Description: Column<StockEntriesRow>;
    InsertDate: Column<StockEntriesRow>;
}

export class StockEntriesColumns extends ColumnsBase<StockEntriesRow> {
    static readonly columnsKey = 'Warehouse.StockEntries';
    static readonly Fields = fieldsProxy<StockEntriesColumns>();
}

[StockEntryStatus]; // referenced types

File: SYP.Web/Modules/Warehouse/StockEntries/StockEntriesColumns.cs

namespace SYP.Warehouse.Columns;

[ColumnsScript("Warehouse.StockEntries")]
[BasedOnRow(typeof(StockEntriesRow), CheckNames = true)]
public class StockEntriesColumns
{
    [EditLink, Width(120)]
    public string EntryNo { get; set; }

    [Width(150)]
    public string WarehouseName { get; set; }

    [Width(150), DisplayFormat("dd.MM.yyyy HH:mm")]
    public DateTime EntryDate { get; set; }

    [Width(100)]
    public StockEntryStatus Status { get; set; }

    [Width(250)]
    public string Description { get; set; }

    [Width(150), DisplayFormat("dd.MM.yyyy HH:mm")]
    public DateTime InsertDate { get; set; }
}
